async/source.go

```diff
@@ -32,59 +56,71 @@ type MessageSource interface {
 // NewMessageSource returns a new insurance asynchronous message source, given
 // an AsyncMessageSource. When Close is called on the AsyncMessageSource,
 // this is also propogated to the underlying AsyncMessageSource.
-func NewMessageSource(ams substrate.AsyncMessageSource) MessageSource {
-	return &messageSourceAdapter{
-		ams,
+func NewMessageSource(source substrate.AsyncMessageSource, opts ...MessageSourceOption) MessageSource {
+	msa := &messageSourceAdapter{
+		source:    source,
+		consumers: 1,
+	}
+
+	for _, opt := range opts {
+		opt(msa)
 	}
+
+	return msa
 }
 
 type messageSourceAdapter struct {
-	ac substrate.AsyncMessageSource
+	source    substrate.AsyncMessageSource
+	msgBuffer uint
+	ackBuffer uint
+	consumers uint
 }
 
 func (a *messageSourceAdapter) ConsumeMessages(ctx context.Context, handler ConsumerMessageHandler) error {
 	rg, ctx := rungroup.New(ctx)
 
-	messages := make(chan substrate.Message)
-	acks := make(chan substrate.Message)
+	messages := make(chan substrate.Message, a.msgBuffer)
```

Comment:
I think by default this would be 0, so it will fail, maybe we can change defaults to 1?

Reply:
The default is 0, yes, which means the channel will be unbuffered https://golang.org/pkg/builtin/#make

Reply:
0 will make a blocking, unbuffered channel. The tests show it works with the default of 0.

Reply:
ah ok, makes sense, ty for explanation